### PR TITLE
Pocket 701 java script ex 2

### PR DESCRIPTION
**Pull request by RuthMcilwaine**

### CODE REVIEW

File: day2/input.js [hunk 1]
```diff
@@ -0,0 +1,1000 @@
...
+up 8
+up 4
+forward 7
+forward 5
+forward 9
+down 6
+forward 7
+forward 9
+up 3
+forward 6`;
```
Comment (tonydengMYOB) [inline day2/input.js:1000]: Can add something like module.exports = { input }; so that it can be used by day2.js

**Comment by tonydengMYOB** [comment]
Good work Ruth. I'm not an expert of JS. But here are a few things you can do with the project:
1. use some error handling/ input validation when you convert string into int
2. try something like module.exports for your input.js so that it can be referenced by your day2.js file
3. try separate your business logic from your **main** class

File: day2/day2.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    obj.direction = output[0];
+    obj.distance = parseInt(output[1]);
+    arr.push(obj);
+  });
+  return arr;
+};
+
+// use global variables without breaking encapsulation
+function main() {
+  let input = `forward 5
```
Comment (NedaOmmani) [inline day2/day2.js:21]: I can't see you use your input.js file. Did you mean to pass that into convertInput?

File: day2/day2.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      case 'up':
+        depth -= value;
+        break;
+      case 'down':
+        depth += value;
+        break;
+      case 'forward':
+        distance += value;
+        break;
+    }
```
Comment (NedaOmmani) [inline day2/day2.js:47]: It's always better to add default: to switch case to be able to handle any unwanted situation.
Imagine there's a typo in the input file, and there is a command like **dowm 2**, How your code would handle 
that?
if you add something like the code below then your code wouldn't crash
default:
break;

File: day2/day2.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const convertInput = (input) => {
+  let command = input.split('\n ');
+  arr = [];
+  command.forEach((element) => {
+    let output = element.split(' ');
+    let obj = {};
+
+    obj.direction = output[0];
+    obj.distance = parseInt(output[1]);
+    arr.push(obj);
```
Comment (NedaOmmani) [inline day2/day2.js:14]: I'd suggest to add some error handling in here so if there are lines which don't follow the expected pattern it wouldn't crash.

Imagine there are lines like these and try to handle them:

- forward a
- 3
- down
- up4
- forward 3up 4

File: day2/day2.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ down 8
+ forward 2`;
+  let converted = convertInput(input);
+  console.log(calculateValue(converted));
+}
+//declare inside function or pass in as arguments
+function calculateValue(arr) {
+  let distance = 0;
+  let depth = 0;
+  arr.forEach((command) => {
```
Comment (NedaOmmani) [inline day2/day2.js:34]: forEach in general is not very efficient so I recommend to mix calculateValue and convertInput to process each line  only once.

**Pull request merged**